### PR TITLE
feat: add producer interceptor interface

### PR DESCRIPTION
**Pull request by ademekici**

### CODE REVIEW

File: producer.go [hunk 1]
```diff
@@ -18,10 +18,11 @@ type Writer interface {
 }
 
 type producer struct {
-	w Writer
+	w           Writer
+	interceptor *ProducerInterceptor
```
Comment (Abdulsametileri) [inline producer.go]: It is not necessary to use interface pointer. I mean `ProducerInterceptor` is interface, no need to use *

File: producer.go [hunk 2]
```diff
@@ -18,10 +18,11 @@ type Writer interface {
...
 }
 
-func NewProducer(cfg *ProducerConfig) (Producer, error) {
+func NewProducer(cfg *ProducerConfig, interceptor *ProducerInterceptor) (Producer, error) {
```
Comment (Abdulsametileri) [inline producer.go]: we can use `interceptor ...ProducerInterceptor` syntax. 
In that way, client can use 
  - NewProducer(producerConfig)
  - NewProducer(producerConfig, interceptor1)
  - NewProducer(producerConfig, interceptor1, interceptor2)

File: producer.go (hunk 1 above)
Comment (Abdulsametileri) [inline producer.go]: I think we can support []ProducerInterceptor, in that way client could implement many interceptors if they want, what do you think?

**Review by Abdulsametileri — CHANGES_REQUESTED** [review]

File: test/stubdata/producer_interceptor_stub.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import "github.com/Trendyol/kafka-konsumer/v2"
+
+type MockProducerInterceptor struct{}
+
+const (
+	XSourceAppKey   = "x-source-app"
+	XSourceAppValue = "kafka-konsumer"
+)
+
+func (i *MockProducerInterceptor) OnProduce(ctx kafka.ProducerInterceptorContext) {
```
Comment (Abdulsametileri) [inline test/stubdata/producer_interceptor_stub.go]: I think, stubdata directory is not necessary, we can put this implementation on integration_test

File: producer_test.go [hunk 4]
```diff
@@ -20,6 +22,26 @@ func Test_producer_Produce_Successfully(t *testing.T) {
...
+		Key:   "x-correlation-id",
+		Value: []byte(utils.UUIDv4()),
+	})
+	interceptor := newMockProducerInterceptor()
+
+	p := producer{w: mw, interceptor: &interceptor}
+
+	// When
+	err := p.Produce(context.Background(), msg)
+	// Then
```
Comment (Abdulsametileri) [inline producer_test.go:39]: Formatting!: We need to put whitespace between 38 and 39

**Review by Abdulsametileri — CHANGES_REQUESTED** [review]

File: producer.go (hunk 1 above)
Comment (ademekici) [reply producer.go]: I fixed hered. You can check 🙏

**Review by Abdulsametileri — APPROVED** [review]
LGTM

**Pull request merged**